src/ImpactEvaluator.sol

```diff
@@ -68,9 +63,11 @@ contract ImpactEvaluator is AccessControl {
     }
 
     function addMeasurements(string memory cid) public virtual returns (uint) {
-        maybeAdvanceRound();
-        emit MeasurementsAdded(cid, currentRoundIndex(), msg.sender);
-        return currentRoundIndex();
+        if (block.number >= currentRoundEnd) {
+            advanceRound();
+        }
+        emit MeasurementsAdded(cid, currentRoundIndex, msg.sender);
```

Comment:
I commented before on the change in the order in which we advance the current round and emit MeasurementsAdded, but I don't remember the answer.

What happens in the current versions:
- spark-api and Station nodes think we are at round N
- Station nodes pick tasks from round N and submit measurements
- we record these measurements as belonging to round N+1, which defines a different set of tasks
- spark-evaluate will reject these measurements because they are for invalid tasks

Unless I am missing something?

I am proposing the following implementation:

```solidity
function addMeasurements(string memory cid) public virtual returns (uint) {
  uint indexOfMeasurements = currentRoundIndex

  if (block.number >= currentRoundEnd) {
    advanceRound();
  }

  emit MeasurementsAdded(cid, indexOfMeasurements, msg.sender);

  return indexOfMeasurements;
}
```

Reply:
The last discussion on this I can find is https://github.com/Meridian-IE/impact-evaluator/pull/26#discussion_r1326041516, which we haven't resolved. I'll create a follow-up PR to revert this part

Reply:
I have moved the emit _before_ `advanceRound`, otherwise we would emit from the previous round after announcing a new round, which is out of order

Reply:
https://github.com/Meridian-IE/impact-evaluator/pull/49